waffle_utils/video/tools.py

```diff
@@ -0,0 +1,163 @@
+
+            if verbose:
+                print(f"{input_path} ({count}) -> {output_path}.")
+
+        success, image = video_capture.read()
+
+    video_capture.release()
+    print(f"Output: {output_dir}/")
+
+
+def create_video(
+    input_dir: Union[str, Path],
+    output_path: Union[str, Path],
+    frame_rate: int = DEFAULT_FRAME_RATE,
+    verbose: bool = False,
+) -> None:
+    f"""Create a Video File from a Directory of Frame Images.
+
+    Args:
+        input_dir (Union[str, Path]): Path to the input directory containing the frame images.
+        output_path (Union[str, Path]): Path to the output video file.
+        frame_rate (int, optional): Frame rate of the output video. Defaults to {DEFAULT_FRAME_RATE}.
+        verbose (bool, optional): Whether to print verbose output. Defaults to False.
+    """
+    input_dir = Path(input_dir)
+    output_path = Path(output_path)
+
+    # Check if all files have the same extension
+    files = list(input_dir.glob("*"))
+    file_extensions = set()
+
+    for file in files:
+        file_extensions.add(file.suffix)
+
+    if len(file_extensions) != 1:
+        raise ValueError(
+            f"The files in {input_dir} do not have a consistent extension."
+        )
+
+    # Check if the extension of the files is supported
+    unique_extension = file_extensions.pop()[1:]  # Get the unique extension existing.
+    if unique_extension not in SUPPORTED_IMAGE_EXTENSION:
+        raise ValueError(
+            f"File extension in {input_dir}: {file_extensions}.\n"
+            "Must be one of {SUPPORTED_IMAGE_EXTENSION}."
+        )
+
+    # Create output directory if it doesn't exist
+    if not output_path.parent.exists():
+        make_directory(input_dir)
+
+    # Get a sorted list of frame image files
+    image_files = natsorted(input_dir.glob(f"*.{unique_extension}"))
+
+    # Load the first frame to get dimensions
+    first_frame = load_image(image_files[0])
+    height, width, _ = first_frame.shape
+
+    # Initialize video writer with the desired codec, frame rate, and frame size
+    output_extension = output_path.suffix
+
+    # Determine the appropriate fourcc codec for the output video format
+    if output_extension == ".mp4":
+        fourcc = cv2.VideoWriter_fourcc(*"mp4v")
+    elif output_extension == ".avi":
+        if cv2.VideoWriter_fourcc(*"MJPG") == -1:
+            fourcc = cv2.VideoWriter_fourcc(*"XVID")
+        else:
+            fourcc = cv2.VideoWriter_fourcc(*"MJPG")
+    elif output_extension == ".wmv":
+        fourcc = cv2.VideoWriter_fourcc(*"WMV2")
+    elif output_extension == ".mov":
+        fourcc = cv2.VideoWriter_fourcc(*"XVID")
+    elif output_extension == ".flv":
+        fourcc = cv2.VideoWriter_fourcc(*"FLV1")
+    elif output_extension == ".mkv":
+        fourcc = cv2.VideoWriter_fourcc(*"VP80")
+    elif output_extension == ".mpeg" or output_extension == ".mpg":
+        fourcc = cv2.VideoWriter_fourcc(*"XVID")
```

Comment:
Use fourcc mapper like this.
```
FOURCC_MAP = {
  "mp4": "mp4v",
  ...
}

Reply:
To handle .avi case, this can also be a solution.
```
FOURCC_MAP = {
  "mp4": cv2.VideoWriter_fourcc(*"mp4v"),
  "avi": cv2.VideoWriter_fourcc(*"XVID") if cv2.VideoWriter_fourcc(*"MJPG") == -1 else cv2.VideoWriter_fourcc(*"MJPG"),
  ...
```

Reply:
Done